services/wireguard/endpoint/endpoint.go

```diff
@@ -99,6 +117,10 @@ func (ce *connectionEndpoint) Stop() error {
 		return err
 	}
 
+	if ce.natService != nil {
```

Comment:
can it be nil ? its depedency

Reply:
connectionEndpoint is used both for provider and consumer. 
There is no reason to do anything with NAT for the consumer.
So the answer is yes, it can be nil for the consumer.

Reply:
That means that connection endpoint is too low level component to handle forwarding. Behaviour based on ifs is not a very good practice

Reply:
Moved NAT configuration to the service manager.